consensus/polybft/bridge_event_relayer.go

```diff
@@ -0,0 +1,460 @@
+	ber.eventTrackers = trackers
+
+	// subscribe relayer to events from the internal chain
+	eventProvider.Subscribe(ber)
+
+	go ber.sendTx()
+
+	return nil
+}
+
+// startTrackerForChain starts a new instance of tracker.EventTracker
+// for listening to the events from an external chain
+func (ber *bridgeEventRelayerImpl) startTrackerForChain(chainID uint64,
+	bridgeCfg *BridgeConfig, runtimeCfg *runtimeConfig) (*tracker.EventTracker, error) {
+	store, err := store.NewBoltDBEventTrackerStore(
+		path.Join(runtimeCfg.DataDir, fmt.Sprintf("/bridge-event-relayer%d.db", chainID)))
+	if err != nil {
+		return nil, err
+	}
+
+	eventTracker, err := tracker.NewEventTracker(
+		&tracker.EventTrackerConfig{
+			EventSubscriber:        ber,
+			Logger:                 ber.logger,
+			RPCEndpoint:            bridgeCfg.JSONRPCEndpoint,
+			SyncBatchSize:          runtimeCfg.eventTracker.SyncBatchSize,
+			NumBlockConfirmations:  runtimeCfg.eventTracker.NumBlockConfirmations,
+			NumOfBlocksToReconcile: runtimeCfg.eventTracker.NumOfBlocksToReconcile,
+			PollInterval:           runtimeCfg.GenesisConfig.BlockTrackerPollInterval.Duration,
+			LogFilter: map[ethgo.Address][]ethgo.Hash{
+				ethgo.Address(bridgeCfg.ExternalGatewayAddr): {
+					bridgeMessageResultEventSig,
+				},
+			},
+		},
+		store,
+		bridgeCfg.EventTrackerStartBlocks[bridgeCfg.ExternalGatewayAddr],
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return eventTracker, eventTracker.Start()
+}
+
+// EventSubscriber implementation
+
+// GetLogFilters returns a map of log filters for getting desired events,
+// where the key is the address of contract that emits desired events,
+// and the value is a slice of signatures of events we want to get.
+// This function is the implementation of EventSubscriber interface
+func (ber *bridgeEventRelayerImpl) GetLogFilters() map[types.Address][]types.Hash {
+	logFilters := map[types.Address][]types.Hash{
+		contracts.BridgeStorageContract: {
+			types.Hash(new(contractsapi.NewBatchEvent).Sig()),
+			types.Hash(new(contractsapi.NewValidatorSetEvent).Sig()),
+		},
+	}
+
+	for _, bridgeCfg := range ber.bridgeConfig {
+		logFilters[bridgeCfg.InternalGatewayAddr] = []types.Hash{types.Hash(
+			new(contractsapi.BridgeMessageResultEvent).Sig())}
+	}
+
+	return logFilters
+}
+
+// ProcessLog is the implementation of EventSubscriber interface,
+// used to handle a log defined in GetLogFilters, provided by event provider
+func (ber *bridgeEventRelayerImpl) ProcessLog(header *types.Header, log *ethgo.Log, dbTx *bolt.Tx) error {
+	var (
+		bridgeMessageResultEvent contractsapi.BridgeMessageResultEvent
+		newBatchEvent            contractsapi.NewBatchEvent
+		newValidatorSetEvent     contractsapi.NewValidatorSetStoredEvent
+	)
+
+	provider, err := ber.blockchain.GetStateProviderForBlock(header)
+	if err != nil {
+		return err
+	}
+
+	systemState := NewSystemState(contracts.EpochManagerContract, contracts.BridgeStorageContract, provider)
```

Comment:
We don't need the `systemState` in case `bridgeMessageResultEvent` arrives. Does it make sense to optimize it (instantiate only when needed)?